pandas/io/parsers.py

```diff
@@ -1571,8 +1572,8 @@ def _get_name(icol):
                 return icol
 
             if col_names is None:
-                raise ValueError(('Must supply column order to use %s as '
-                                  'index') % str(icol))
+                raise ValueError(('Must supply column order to use {icol!s} '
```

Comment:
Do we need to be explicit about type in this and subsequent example?

Reply:
if you said about '!s', yes. !s is replacing the str function not '%s'.